lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -664,7 +671,7 @@ insertCheckpoint wid cp = do
     insert_ cp'
     putTxMetas metas
     putTxs ins outs
-    dbChunked insertMany_ utxo
+    mapM_ (insertWhenNew wid) utxo
```

Comment:
You should be able to replace this with `mapM_ insertUniqueEntity utxo`. It will be faster than insertWhenNew. Hopefully the db bench will show us how much putting checkpoints slows down with the rollback changes.

Reply:
if we do not change this block then we will have for the same UTxO a number of rows for slots slot0, slot1, slot2,... insertWhenNew makes sure we insert once only. Thanks to that we will have slot0..when putTxHistory we will change spent_utxo from Nothing to (just slotSpent) within the same row. If I understand well this is what we want, no?

Reply:
What happens in persistent if we try to insert an entity that already exists? Isn't there any sort of `tryInsert` where we can try to insert something, and silently ignore the query if the entity already exists.. ?

Reply:
:point_up: seems like `upsert` would work for this? It inserts if not exists, and we can simply have an identity as the udpate function if it exists.

Reply:
Looking at UTxO table we have : 
```
158     Primary                                                                                                                                                                               
159         utxoWalletId                                                                                                                                                                      
160         utxoSlot                                                                                                                                                                          
161         utxoInputId                                                                                                                                                                       
162         utxoInputIndex                                                                                                                                                                    
163         utxoOutputAddress                                                                                                                                                                 
164         utxoOutputCoin
```
meaning if I have one utxo row with let's say `utxoSlot=slot1` . Now I do not to want to insert when `utxoSlot=slot3` with other column values the same. So I need to check if the row for all column values beside utxoSlot is existent. If not then insert, otherwise leave it

Reply:
tried something like that : 
```
dbChunked insertWhenNew1 utxo

781  insertWhenNew1                                                                                                                                                                           
782      :: [UTxO]                                                                                                                                                                            
783      -> SqlPersistT IO ()                                                                                                                                                                 
784  insertWhenNew1 utxos = do                                                                                                                                                                
785      mapM_ insertUniqueEntity utxos
```
but failed;-/ more utxos were inserted

Reply:
nevertheless, refactored the current version to be used with `dbChunked`

Reply:
`insertUniqueEntity` _is_ supposed to be the "try insert and ignore failure" function that we want -- at least according to my reading of the persistent docs. I will investigate.

Reply:
I have changed it to use `dbChunked putMany utxos`. It's the best option I could find in the persistent docs.

`dbChunked` is not necessary when using `mapM_` and inserting _single_ records.

But we want to use the batch insert operations if at all possible.

We need to handle the case of conflicting utxo because it is inserting the entire wallet utxo every checkpoint. It would be nice to be able to avoid that somehow, but that's how it is now.